spec/large_object_store_spec.rb

```diff
@@ -35,6 +35,18 @@ def delete(key)
   end
 end
 
+RSpec::Matchers.define :be_compressed do
+  match do |actual|
+    actual[LargeObjectStore::UUID_SIZE] == 'z' || actual[0] == 'z'
```

Comment:
I put the compression indicator byte after your uuid, not before it.
However, you prepend slices with the uuid when it's a multiple page entry, but not for single page entries.
For single page entries, the 0/z is in position 0. For multiple page entries, it's in position UUID_SIZE on the first page.